### PR TITLE
Add JDK 17 instructions to Maven page

### PR DESCRIPTION
**Pull request by sarahhaggarty**
This PR updates the Maven page by adding instructions how to use the latest LTS JDK version 17. It also tidies up the page with more consistent headings and minor grammar fixes.

This PR resolves: [KT-56137](https://youtrack.jetbrains.com/issue/KT-56137/Maven-Explain-how-to-use-Kotlin-with-the-latest-LTS-JDK-version).

### CODE REVIEW

**Review by Tapchicoma — APPROVED** [review]

File: docs/topics/maven.md [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+```none
+--add-opens=java.base/java.lang=ALL-UNNAMED
+--add-opens=java.base/java.io=ALL-UNNAMED
+```
+
+## Set dependencies
 
 Kotlin has an extensive standard library that can be used in your applications.
-To use the standard library in your project, add the following dependency in the pom file:
+To use the standard library in your project, add the following dependency in your `pom.xml` file:
```
Comment (CherepanovAleksei) [inline docs/topics/maven.md:29]: NIT: Should we use "**to** your `pom.xml` file:" here?

**Review by CherepanovAleksei — APPROVED** [review]

File: docs/topics/maven.md (hunk 1 above)
Comment (sarahhaggarty) [reply docs/topics/maven.md]: Nice suggestion. Fixed!

File: docs/topics/maven.md [hunk 2]
```diff
@@ -326,14 +336,12 @@ write a dependency on [`kotlin-bom`](https://mvnrepository.com/artifact/org.jetb
...
-formats, including standard Javadoc.
+The standard Javadoc generation plugin (`maven-javadoc-plugin`) doesn't support Kotlin code. To generate documentation 
+for Kotlin projects, use [Dokka](https://github.com/Kotlin/dokka). For more information, see [Run Dokka – Maven](dokka-maven.md).
+Dokka supports mixed-language projects and can generate output in multiple formats, including standard Javadoc.
 
-## OSGi
+## Enable OSGi support
 
-For OSGi support see the [Kotlin OSGi page](kotlin-osgi.md).
+See [Kotlin and OSGi – Maven](kotlin-osgi.md#maven).
```
Comment (koshachy) [inline docs/topics/maven.md:347]: Not sure that we use such notation for links.

Maybe:
Learn how to enable OSGi support in your Maven project?

**Review by koshachy — APPROVED** [review]
LGTM,
See the comment, please.

File: docs/topics/maven.md (hunk 2 above)
Comment (sarahhaggarty) [reply docs/topics/maven.md]: We never agreed on a convention 😄 I'll update with your suggestion 👍

**Pull request merged**